src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -669,7 +669,7 @@ private void retainAutomaticFeedbacksWithTestCase(Result result, final Set<Progr
                 && testCases.stream().noneMatch(test -> test.getTestName().equalsIgnoreCase(feedback.getText())));
 
         // If there are no feedbacks left after filtering those not valid, also setHasFeedback to false.
-        if (result.getFeedbacks().stream().noneMatch(feedback -> Boolean.FALSE.equals(feedback.isPositive())
+        if (result.getFeedbacks().stream().noneMatch(feedback -> Boolean.TRUE.equals(feedback.hasDetails())
                 || feedback.getType() != null && (feedback.getType().equals(FeedbackType.MANUAL) || feedback.getType().equals(FeedbackType.MANUAL_UNREFERENCED)))) {
```

Comment:
```suggestion
        if (result.getFeedbacks().stream().noneMatch(Feedback::hasDetails) || feedback.getType() != null && (feedback.getType().equals(FeedbackType.MANUAL) || feedback.getType().equals(FeedbackType.MANUAL_UNREFERENCED)))) {
```
Thanks, for fixing this piece of code, too.
One suggestion before approving: The `Boolean.TRUE` is not necessary here as `hasFeedback` returns a `boolean` already.